django/contrib/gis/gdal/raster/source.py

```diff
@@ -72,15 +134,35 @@ def __repr__(self):
         """
         Short-hand representation because WKB may be very large.
         """
-        return '<Raster object at %s>' % hex(addressof(self.ptr))
+        return '<Raster object at %s>' % hex(addressof(self._ptr))
+
+    def _flush(self):
+        """
+        Flush all data from memory into the source file if it exists.
+        The data that needs flushing are geotransforms, coordinate systems,
+        nodata_values and pixel values. This function will be called
+        automatically wherever it is needed.
+        """
+        # Raise an Exception if the value is being changed in read mode.
+        if not self._write:
+            raise GDALException(
+                'Raster needs to be opened in write mode to change values.')
```

Comment:
Added an exception here, I realized that GDAL does not throw errors when changes are made on files opened in read mode, so this has to be checked on the python side.